deflate_medium.c

```diff
@@ -273,15 +273,15 @@ ZLIB_INTERNAL block_state deflate_medium(deflate_state *s, int flush) {
         /* move the "cursor" forward */
         s->strstart += current_match.match_length;
 
-        if (bflush)
+        if (UNLIKELY(bflush))
             FLUSH_BLOCK(s, 0);
     }
     s->insert = s->strstart < MIN_MATCH-1 ? s->strstart : MIN_MATCH-1;
     if (flush == Z_FINISH) {
```

Comment:
Why not `UNLIKELY` here too?

Reply:
It hurts short compressions, and also hurts performance in my huge-file tests. As I mentioned, deflate_medium needed a lot of testing.

Reply:
I wonder about doing: s->insert = UNLIKELY(s->strstart < MIN_MATCH-1) ? s->strstart : MIN_MATCH-1;